.github/workflows/main.yml

```diff
@@ -151,3 +145,57 @@ jobs:
           cache-to: type=gha,mode=max
           build-args: |
             TARGET_GOLANG_VERSION=${{ env.TARGET_GOLANG_VERSION }}
+
+
+  # Run e2e tests on devnet IF the PR has a label "e2e-devnet-test"
+  e2e-tests:
```

Comment:
I missed a clarification in the Github Actions documentation that this would only work on the default branch, so having a separate workflow did not work. Going to have a larger main workflow instead.